plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -351,6 +363,28 @@ public static ReaderPageSource createPageSource(
         return Optional.of(new GroupType(baseType.getRepetition(), baseType.getName(), ImmutableList.of(type)));
     }
 
+    private static Optional<ColumnIndexStore> getColumnIndexStore(
+            Predicate parquetPredicate,
+            ParquetDataSource dataSource,
+            BlockMetaData blockMetadata,
+            Map<List<String>, RichColumnDescriptor> descriptorsByPath,
+            ParquetReaderOptions options)
+    {
+        if (!options.isUseColumnIndex() || !(parquetPredicate instanceof TupleDomainParquetPredicate)) {
```

Comment:
This code shouldn't be aware of the type of `Predicate`, so remove the `!(parquetPredicate instanceof TupleDomainParquetPredicate)` part.